runtime/near-vm-runner/src/wasmtime_runner.rs

```diff
@@ -388,6 +373,43 @@ enum PreparationResult {
     Ready(ReadyContract),
 }
 
+enum RunOutcome {
+    Ok,
+    AbortNop(FunctionCallError),
+    Abort(FunctionCallError),
+}
```

Comment:
Was this enum necessary because of lifetimes on logic? Maybe it would be worth it to add a brief comment to that effect?

Reply:
I initially tried passing `logic` in as a parameter and rely on `VMOutcome::abort(logic.result_state, error)` and such, however that caused `logic` to be dropped earlier than it should, since it would need to be moved.
I'll add a comment